main/integration-tests/README.md

```diff
@@ -1,66 +1,90 @@
+Use the same config values used in `/main/config/settings` for fields awsRegion, awsProfile, solutionName, envName, ane envType. 
 
 **Note**
 
-This file is unique from the rest of the `<STAGE>.yml` files created in the SDCs in that it does not gather serverless settings passed on from the hierarchies above (eg. from `main/config/settings/.defaults.yml`)
+This file is unique from other `<STAGE>.yml` files under other SDCs. It does not gather serverless settings passed on from the hierarchies above (eg. from `main/config/settings/.defaults.yml`)
+
+## Execution
 
-## Execution:
+Once test resources and config file are created, you can run the integration tests against the dev environment defined in the config file.
 
-Now run the command below to trigger the integration test suite. Note that this will generate test-related resources in the deployment linked to the provided stage name.
+Note: Integration tests will create resources in the environment they executes against.
 
+### Run against dev environment
+- In config file `main/integration-tests/config/settings/<STAGE>.yml`
+  - set `isBuildServer` to `false`
+  - set `isLocal` to `false`
+ 
+run all integration tests from the root directory with: 
 ```bash
 $ scripts/run-integration-tests.sh <STAGE>
 ```
 
-If you want to run a specific test suite, you can use the following command:
-
+Run specific test suites under `main/integration-tests` with: 
 ```bash
-$ cd integration-tests
 $ pnpm intTest __test__/api-tests/<your test suite file> -- --stage=<STAGE>
 # IMPORTANT: notice the additional '-- ' in front of the '--stage='
 ```
+
+Aside from running against a dev environment, integration tests can also be ran against local deployment, in a CI/CD Pipeline, or through a GitHub workflow. 
+
+### Run against local deployment (SLS Offline)
+- In config file `main/integration-tests/config/settings/<STAGE>.yml`
+  - set `isBuildServer` to `false`
+  - set `isLocal` to `true`
+  - set `localApiEndpoint` to local API endpoint
+- Launch sls offline
+- Trigger integration tests with the same commands
+
+### Run in CI/CD Pipeline
+- In config file `main/integration-tests/config/settings/<STAGE>.yml`
+  - set `isBuildServer` to `true`
+  - set `isLocal` to `false`
+  - set `awsProfile` to the AWS account used for integration tests
+- Follow the steps mentioned in the `main/cicd/README.md` file to set up the CI/CD Pipeline
+- The integration test config file will be automatically saved in the deployment S3 bucket `<namespace>-artifacts` under `integration-test` folder
+- Integration test will be triggered as part of the CI/CD pipeline
+
+### Run in GitHub Workflow
+- In config file `main/integration-tests/config/settings/<STAGE>.yml`
+  - set `isBuildServer` to `true`
+  - set `isLocal` to `false`
+- Upload the config file to the deployment S3 bucket `<namespace>-artifacts` under `integration-test` folder
```

Comment:
Might be worth mentioning that the upload step is only required if the file is already not present. (It is possible they have a CI/CD pipeline as well that pushed the config file to its S3 location)